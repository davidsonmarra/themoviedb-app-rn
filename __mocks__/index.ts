import MovieDTO from '../src/@types/movies-dto';

export const mockCardMovie: MovieDTO = {
  id: 1,
  title: 'Movie Title',
  overview: 'Movie Overview',
  poster_path: '7qOSKoOAPgemYhBwbJgBWcCxPWZ.jpg',
  release_date: '2021-01-01',
  backdrop_path: 'path/to/image',
  vote_average: 10,
  genre_ids: [1, 2],
};

export const mockListMovie: MovieDTO[] = [
  {
    id: 1,
    title: 'Movie Title',
    overview: 'Movie Overview',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 10,
    genre_ids: [1, 2],
  },
  {
    id: 2,
    title: 'Movie Title 2',
    overview: 'Movie Overview 2',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 9,
    genre_ids: [1, 2],
  },
];
