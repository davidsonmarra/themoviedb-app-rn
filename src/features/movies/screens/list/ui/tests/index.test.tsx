import React from 'react';
import ListContainer from '..';
import {render, screen} from '@testing-library/react-native';
import {mockCardMovie, mockListMovie} from '../../../../../../../__mocks__';

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

const containerInstance = (props: any) => render(<ListContainer {...props} />);

describe('ListContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({
      moviesData: mockCardMovie,
      loadingFetchMovies: false,
      onEndReached: jest.fn(),
      onPressMovie: jest.fn(),
      onSearch: jest.fn(),
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should render movie details', () => {
    containerInstance({
      moviesData: mockListMovie,
      loadingFetchMovies: false,
      onEndReached: jest.fn(),
      onPressMovie: jest.fn(),
      onSearch: jest.fn(),
    });
    const title = screen.getByText(mockCardMovie.title);
    expect(title).toBeTruthy();
  });

  it('should call onEndReached when end of list is reached', () => {
    const onEndReached = jest.fn();
    containerInstance({
      moviesData: mockListMovie,
      loadingFetchMovies: false,
      onEndReached,
      onPressMovie: jest.fn(),
      onSearch: jest.fn(),
    });
    const flatList = screen.getByTestId('movies-flatlist');
    flatList.props.onEndReached();
    expect(onEndReached).toHaveBeenCalled();
  });

  it('should call onSearch when search icon is pressed', () => {
    const onSearch = jest.fn();
    containerInstance({
      moviesData: mockListMovie,
      loadingFetchMovies: false,
      onEndReached: jest.fn(),
      onPressMovie: jest.fn(),
      onSearch,
    });
    const searchIcon = screen.getByTestId('right-buton-icon');
    searchIcon.props.onClick();
    expect(onSearch).toHaveBeenCalled();
  });
});
