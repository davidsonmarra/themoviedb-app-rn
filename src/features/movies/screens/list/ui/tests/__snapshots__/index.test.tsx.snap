// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer should render correctly 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "#01B4A4",
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="back-button"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="#FFFFFF"
        focusable={false}
        height="24"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="24"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          {
            "flex": 1,
            "textAlign": "center",
          },
        ]
      }
    >
      Listagem
    </Text>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="right-buton-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="#FFFFFF"
        focusable={false}
        height="24"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="24"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M784-120L532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56zM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    contentContainerStyle={
      {
        "padding": 16,
        "paddingBottom": 100,
      }
    }
    data={
      {
        "backdrop_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
        "genre_ids": [
          1,
          2,
        ],
        "id": 1,
        "overview": "Movie Overview",
        "poster_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
        "release_date": "2021-01-01",
        "title": "Movie Title",
        "vote_average": 10,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    keyboardDismissMode="on-drag"
    onContentSizeChange={[Function]}
    onEndReached={[MockFunction]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    testID="movies-flatlist"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <ActivityIndicator
          color="#01B4A4"
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
